### ISSUE TITLE
Iteration 11 Holistic Feedback

### ISSUE THREAD

**Issue by CDR-API-Stream** [ConsumerDataStandardsAustralia/standards-maintenance#511]
# Description
This CR has been created to simplify the raising of minor changes, such as text corrections or description clarifications, that are not really material to the standards but do have real impact of readability and clarity.

Please raise any such suggestions that you would like included in Maintenance Iteration 11 and the DSB will review them.  If we believe the suggestion are material we will raise a dedicated CR for them.

**Comment by perlboy** [comment]
First comment, this one looks like it could be rolled in: https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/461

**Comment by perlboy** [comment]
Clarification on `Get Agreed Payment Schedule` sought during implementation calls has indicated that 404 is not a valid response for situations where an account has no specifically agreed payment schedule and that instead when a Consumer "signs up" to a plan they agree to a payment schedule.

This leads to the following clarifications required:

1. The description seems like it should be altered to _"Obtain the agreed payment schedule and details for a specific energy account"_ as "if any" implies conditionality
2. What `manualPayment` and `cardDebit` responses look like for situations where the consumer has paid in advance (ie. they get a bill but require no payment)
3. What the Agreed Payment Schedule is for prepaid energy plans (such as [Reamped Advance](https://www.reampedenergy.com.au/our-price-plans/act-nsw-sa-qld/)

There is also an MI item https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/495 for this, happy to move this comment there if it seems more appropriate.

**Comment by perlboy** [comment]
`EnergyBillingPaymentTransaction.method` refers to card payments as `CARD` while `EnergyPlanContract.paymentOption` refers to `CREDIT_CARD`. Given both are about acquisition method suggest they be aligned to simply `CARD`.

**Comment by perlboy** [comment]
`EnergyServicePointDetail.timeOfDay` doesn't seem to include spaces in enums, notably `ALLDAY` and `OFFPEAK`. `OFFPEAK` in particular is notable because there is `OFF_PEAK` in other payloads too.

**Comment by perlboy** [comment]
`EnergyPlanGreenPowerCharges.scheme` doesn't seem to include spaces in enums, `GREENPOWER` should probably be `GREEN_POWER`

**Comment by perlboy** [comment]
Energy Billing systems allow for _multiple_ payment schedules while the payment schedules endpoint only allows for a single value. Clarification on how this should be handled is requested (in addition to the other elements regarding no explicit payment schedule).

**Comment by perlboy** [comment]
The following fields are currently specified in the Standards as `number` but examples are an `integer` and they appear to actually be integers in [source](https://www.aemo.com.au/-/media/Files/Stakeholder_Consultation/Consultations/NEM-Consultations/2019/DER-register/Final/DER-Register-Information-Guidelines.pdf) [specifications](https://aemo.com.au/-/media/files/electricity/nem/it-systems-and-change/2020/der-register-release-schedule-and-technical-specification.pdf) or [underlying standard](https://store.standards.org.au/product/as-nzs-4777-2-2020):
- `EnergyDerRecord.availablePhasesCount`: Phases are always whole numbers and according to DER Guidelines it is an enumeration ("Pick List") of `[1,2,3]` so should probably be an enumeration
- `EnergyDerRecord.installedPhasesCount`: Phases are always whole numbers and according to DER Guidelines it is an enumeration ("Pick List") of `[1,2,3]` so should probably be an enumeration
- `EnergyDerRecord.protectionMode.exportLimitkva`: This field also appears to be incorrectly camelCased as KVA and kVA are interchangeable and KVA is used elsewhere, would be best to be consistent

The following fields have implied defaults if not specified and DER Guidelines state `Default values AS4777-2: 2015 section 7.4` which is a paid document so having a description that provides them would be useful:
- `EnergyDerRecord.protectionMode.underFrequencyProtection`: AS4777-2 default is 47Hz (Aus A/B) or 45Hz (Aus C)
- `EnergyDerRecord.protectionMode.underFrequencyProtectionDelay`: AS4777-2 default is 1s (Aus A/B) or 5s (Aus C)
- `EnergyDerRecord.protectionMode.overFrequencyProtection`: AS4777-2 default is 52Hz (Aus A/B) or 55Hz (Aus C)
- `EnergyDerRecord.protectionMode.overFrequencyProtectionDelay`: AS4777-2 has this marked as `-` and it is missing in the DER Guidelines but exists in the DER Register Technical specification
- `EnergyDerRecord.protectionMode.underVoltageProtection`: AS4777-2 default is 180V
- `EnergyDerRecord.protectionMode.underVoltageProtectionDelay`: AS4777-2 default is 10s
- `EnergyDerRecord.protectionMode.overVoltageProtection`: AS4777-2 default is 265V
- `EnergyDerRecord.protectionMode.overVoltageProtectionDelay`: AS4777-2 default is 1s
- `EnergyDerRecord.protectionMode.sustainedOverVoltage`: AS4777-2 default is 275V

The following fields seem either non-existent or incorrectly defined:
- `EnergyDerRecord.protectionMode.sustainedOverVoltageDelay`: There is no trip delay specified for `V>>` in the DER Guidelines and AS4777-2 has these marked as `-`. The DER Register Technical specification _has_ this field but it is described as "Sustained Over voltage protection delay in volts (V)" which seems quite wrong (it's a time measurement not a voltage). It's unclear if this field even exists?
- `EnergyDerRecord.acConnections[].derDevices[].manufacturer` appears to be mandatory and in fact an enumeration ("pick list") in the DER Guidelines with a common of _"Definitions align to the approved modules list."_.
- `EnergyDerRecord.acConnections[].derDevices[]. modelNumber ` appears to be mandatory and in fact an enumeration ("pick list") in the DER Guidelines with a common of _"Definitions align to the approved modules list."_

The following fields have descriptions worth that should probably be fixed with proposals in italics:
- `EnergyDerRecord.acConnections[].status`: _**Enumeration** used to indicate the status of the Inverter. This will be used to identify if an inverter is ACTIVE or INACTIVE or DECOMMISSIONED_
- `EnergyUsageRead.intervalRead.intervalReads`: _... specified by readIntervalLength beginning at midnight (AEST) of readStartDate ..._

**Comment by perlboy** [comment]
Clarification is requested for `oldest-date` and `newest-date` filters:
- `oldest-date` filter specifies the default value is _"If absent defaults to newest-date minus 24 months days"_. Is this a simple 24 months subtraction (ie. "today" is 2022-05-15 so `oldest-date` would be `2020-05-15`) that ignores other conditions (length of months, leap years etc) or something more complex?
- What timezone are `oldest-date` and `newest-date` to be considered at? UTC seems feasible but clarification is important relative to `intervalReads` (see below)
- `EnergyUsageRead.intervalRead.intervalReads` specifies `specified by readIntervalLength beginning at midnight of readStartDate`. What is "midnight"? I _think_ this is AEST however making it explicit in the description would be helpful to implementers

**Comment by JamesMBligh** [comment]
Via CDR Support Portal.  The language for some of the common types (e.g. MaskedPANString) use 'should' inconsistently with RFC2119 as their language predates the use of the RFC.  The type descriptions should be reviewed to align to the intent (i.e. make each 'should' an explicit **SHOULD** or **MUST**).

**Comment by nils-work** [comment]
From v1.17.0, I think the anchor link in this section is incorrect, perhaps it should point to -
https://consumerdatastandardsaustralia.github.io/standards/#profile-scope-and-standard-claims

![image](https://user-images.githubusercontent.com/60594671/169948537-f9804db1-9ded-4980-8f3c-bf841cb327f7.png)

**Comment by markverstege** [comment]
The following CRs have been included in Maintenance Iteration 11. Each represents a minor documentation fix. They've been cross posted here within the holistic feedback CR for the iteration. Changes will be staged against each within a maintenance/511 branch as individual commits.

* [#487: DCR APIs non-normative examples would benefit from clarification](https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/487)
  Action: Fix non-normative example
* [#492: Admin API Definitions Request Body Incorrectly Nested](https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/492)
  Action: fix Admin API Swagger
* [#494: Response payload structure description error](https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/494)
  Action: fix documentation
* [#489: v1.15.0 More ambiguity into x-fapi-auth-date not less](https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/489)
    Action: Change Banking API endpoint descriptions to be:
```
The time when the customer last logged in to the Data Recipient Software Product as described in [FAPI-R-Draft]. 
Required for all resource calls (customer present and unattended). Not required for unauthenticated calls.
```

The following CRs will be closed:
* [#78: HTTP Header to be returned in the case where the request is not entirely well formed and a large page size is requested](https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/78)
  Action: Issue has been **closed**
* [#200: Documentation defect in the non-normative example - DH calling the DR Revocation Endpoint](https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/200)
  Action: **Closed with no change** at the end of MI 11 if no further feedback is received

**Comment by CDR-API-Stream** [comment]
> From v1.17.0, I think the anchor link in this section is incorrect, perhaps it should point to - https://consumerdatastandardsaustralia.github.io/standards/#profile-scope-and-standard-claims
> 
> ![image](https://user-images.githubusercontent.com/60594671/169948537-f9804db1-9ded-4980-8f3c-bf841cb327f7.png)

This issue has been fixed. It is staged for review here: https://github.com/ConsumerDataStandardsAustralia/standards-staging/compare/release/1.18.0...maintenance/511. [Commit cb9e55d](https://github.com/ConsumerDataStandardsAustralia/standards-staging/commit/cb9e55d73cdaf307583cfbae43211276988d515f).

**Comment by CDR-API-Stream** [comment]
> It seems like these two `*` points may have been intended to be bullet points -
> ![image](https://user-images.githubusercontent.com/60594671/169188365-f475ce52-bbfe-4bf7-8ae1-f31d6bc2a667.png)

This issue has been fixed. It is staged for review here: https://github.com/ConsumerDataStandardsAustralia/standards-staging/compare/release/1.18.0...maintenance/511. [Commit 7b05677](https://github.com/ConsumerDataStandardsAustralia/standards-staging/commit/7b05677c8386872e19e4debb39218b37011df016).

**Comment by CDR-API-Stream** [comment]
Issue #487 has been staged for review: https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/487#issuecomment-1136626910

**Comment by markverstege** [comment]
To improve consistency, **bold** all Requirement Level keywords usages (e.g. **MUST**, **SHOULD**, ...) throughout the standards.

**Comment by CDR-API-Stream** [comment]
Issue #494 has been staged for review: https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/494#issuecomment-1136671959

**Comment by CDR-API-Stream** [comment]
> # Description
> This CR has been created to simplify the raising of minor changes, such as text corrections or description clarifications, that are not really material to the standards but do have real impact of readability and clarity.
> 
> Please raise any such suggestions that you would like included in Maintenance Iteration 11 and the DSB will review them. If we believe the suggestion are material we will raise a dedicated CR for them.

The baseline review branch for these holistic changes has been staged here: https://github.com/ConsumerDataStandardsAustralia/standards-staging/compare/release/1.18.0..maintenance/511. As changes are addressed in the comments above, they will be incorporated into this staging branch for community review.

**Comment by CDR-API-Stream** [comment]
Issue #489 has been staged for review: https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/489#issuecomment-1136697800

**Comment by HemangCDR** [comment]
Specify that the sort order of usage reads in the energy standards (i.e. in [EnergyUsageListResponse](https://consumerdatastandardsaustralia.github.io/standards/#tocSenergyusagelistresponse)) will be by NMI (ascending), read start date (descending)

**Comment by CDR-API-Stream** [comment]
Issue #497 has been staged for review: https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/497#issuecomment-1138090708

**Comment by perlboy** [comment]
Sort order of usage data is not specified. In fact the [DP for usage data](https://github.com/ConsumerDataStandardsAustralia/standards/issues/195) didn't seem to specify it either. It appears to have last been discussed in an abandoned [prior decision proposal](https://github.com/ConsumerDataStandardsAustralia/standards/issues/112).

**Comment by HemangCDR** [comment]
@perlboy no worries.

DP 195 did have the sort order specified, it was in the "General Notes" section of the proposal as below:
- _"Reads will be sorted by NMI followed by read start date in descending order"_

It was missed in the decision document (which is probably why it isn't in the standards). We will ratify it part of MI11.

**Comment by perlboy** [comment]
> @perlboy no worries.
> DP 195 did have the sort order specified, it was in the "General Notes" section of the proposal as below:
> * _"Reads will be sorted by NMI followed by read start date in descending order"_
> It was missed in the decision document (which is probably why it isn't in the standards). We will ratify it part of MI11.

Yes. That was what I was highlighting. The only thing binding is the Standards which are tied to decisions of the Chair. Consequently the sort order isn't _actually_ ratified by the Chair. It should be. 😄

**Comment by nilsberge** [comment]
Another minor documentation update similar to [above](#issuecomment-1136597967), I think these should also be a list -
![image](https://user-images.githubusercontent.com/1802516/170480096-b605a40a-05d0-4481-b261-0860f541318d.png)

**Comment by OpalRussAEMO** [comment]
Suggested updates to EnergyDerRecord, note some of these overlap with Stuart's suggestions.

- EnergyDerRecord.availablePhasesCount - should be a positive integer
- EnergyDerRecord.installedPhasesCount - should be a positive integer
- EnergyDerRecord.acConnections.derDevices.count - should be a positive integer
- EnergyDerRecord.islandableInstallation - should be a boolean not a string. It is a yes/no question.

In regards to Stuart's comments on the manufacturer and model's potentially being enumerated, these values aren't enumerated in the DER Register so enumeration can't be supported.

**Comment by OpalRussAEMO** [comment]
> Clarification is requested for `oldest-date` and `newest-date` filters:
> 
> * `oldest-date` filter specifies the default value is _"If absent defaults to newest-date minus 24 months days"_. Is this a simple 24 months subtraction (ie. "today" is 2022-05-15 so `oldest-date` would be `2020-05-15`) that ignores other conditions (length of months, leap years etc) or something more complex?
> * What timezone are `oldest-date` and `newest-date` to be considered at? UTC seems feasible but clarification is important relative to `intervalReads` (see below)
> * `EnergyUsageRead.intervalRead.intervalReads` specifies `specified by readIntervalLength beginning at midnight of readStartDate`. What is "midnight"? I _think_ this is AEST however making it explicit in the description would be helpful to implementers

- We're using a simple subtraction of 24 months unless advised otherwise
- All dates/times relating to meter data and standing data (service point detail) are recorded in AEST therefore all dates/times for meter data should be interpreted as AEST.

- EnergyUsageRead.meterID: for consistency, should this be meterId (lowercase d)?
- EnergyUsageRead.readStartDate: can we add more to the description.  e.g. For intervals, this date should be interpreted as of 12:00 am AEST.  
- EnergyUsageRead.readEndDate: can we note that this date will only be provided for basic reads and update the description to remove reference to time?  For intervals, the reads represent a single day of intervals captured starting at midnight AEST therefore readEndDate won't be provided.
- EnergyUsageRead.unitOfMeasure: can we note that values will be provided in uppercase?

**Comment by CDR-API-Stream** [comment]
> Another minor documentation update similar to [above](#issuecomment-1136597967), I think these should also be a list -
> ![image](https://user-images.githubusercontent.com/1802516/170480096-b605a40a-05d0-4481-b261-0860f541318d.png)

This issue has been fixed. It is staged for review here: https://github.com/ConsumerDataStandardsAustralia/standards-staging/compare/release/1.18.0...maintenance/511. [Commit 4e285de](https://github.com/ConsumerDataStandardsAustralia/standards-staging/commit/4e285de008f29b8a0e631ed537bdbb184441d81a).

**Comment by CDR-API-Stream** [comment]
> Clarification on `Get Agreed Payment Schedule` sought during implementation calls has indicated that 404 is not a valid response for situations where an account has no specifically agreed payment schedule and that instead when a Consumer "signs up" to a plan they agree to a payment schedule.
> 
> This leads to the following clarifications required:
> 
> 1. The description seems like it should be altered to _"Obtain the agreed payment schedule and details for a specific energy account"_ as "if any" implies conditionality
> 2. What `manualPayment` and `cardDebit` responses look like for situations where the consumer has paid in advance (ie. they get a bill but require no payment)
> 3. What the Agreed Payment Schedule is for prepaid energy plans (such as [Reamped Advance](https://www.reampedenergy.com.au/our-price-plans/act-nsw-sa-qld/) or [Powershop Powerpacks](https://www.powershop.com.au/why-powershop/powershop-experience/powerpacks/#/))
> 4. Energy Billing systems allow for _multiple_ payment schedules while the payment schedules endpoint only allows for a single value. How should a retailer present this?
> 5. What the response should look like if it is a PayPal payment schedule?
> 
> There is also an MI item #495 for this, happy to move this comment there if it seems more appropriate.
> 
> Further update from implementation call for this endpoint:
> 
> * `isTokenised` says `False if absent. If false then bsb and accountNumber should not be expected to be included` but `bsb`/`accountNumber` says `Is required if isTokenised is absent or false` which seems to be conflicting
>   
>   * (@perlboy note) This one was added when a representative from a Retailer raised it but does seem to be aligned in the sense if `isTokenised` is `true` then `bsb`/`accountNumber` would not be expected. The wording though seems to be a double negative. Suggest the following:
>     
>     1. `isTokenised` phrase stated as `False if absent. If true then bsb and accountNumber must be included`
>     2. `bsb` and `accountNumber` phrase left unchanged or possibly removed entirely leaving `isTokenised` qualifier as the source of truth

The above comments have been addressed in CR #495. Please refer to this [comment](https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/495#issuecomment-1151897922).

**Comment by CDR-API-Stream** [comment]
In response to @perlboy's [comment](https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/511#issuecomment-1126897759)
> The following fields are currently specified in the Standards as `number` but examples are an `integer` and they appear to actually be integers in [source](https://www.aemo.com.au/-/media/Files/Stakeholder_Consultation/Consultations/NEM-Consultations/2019/DER-register/Final/DER-Register-Information-Guidelines.pdf) [specifications](https://aemo.com.au/-/media/files/electricity/nem/it-systems-and-change/2020/der-register-release-schedule-and-technical-specification.pdf) or [underlying standard](https://store.standards.org.au/product/as-nzs-4777-2-2020):
> 
> * `EnergyDerRecord.availablePhasesCount`: Phases are always whole numbers and according to DER Guidelines it is an enumeration ("Pick List") of `[1,2,3]` so should probably be an enumeration
> * `EnergyDerRecord.installedPhasesCount`: Phases are always whole numbers and according to DER Guidelines it is an enumeration ("Pick List") of `[1,2,3]` so should probably be an enumeration

Based on @OpalRussAEMO's [comment](https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/511#issuecomment-1144297476), the DSB proposes to make the above attributes PositiveInteger.  We can update the description to state the acceptable values are 1, 2 and 3.



> * `EnergyDerRecord.protectionMode.exportLimitkva`: This field also appears to be incorrectly camelCased as KVA and kVA are interchangeable and KVA is used elsewhere, would be best to be consistent

The above attribute is represented exactly as it is from AEMOs DER specification similar to other attributes.

> The following fields have implied defaults if not specified, may be integers and DER Guidelines state `Default values AS4777-2: 2015 section 7.4` which is a paid document so at a minimum having a description that provides them would be useful:
> 
> * `EnergyDerRecord.protectionMode.underFrequencyProtection`: AS4777-2 default is 47Hz (Aus A/B) or 45Hz (Aus C)
> * `EnergyDerRecord.protectionMode.underFrequencyProtectionDelay`: AS4777-2 default is 1s (Aus A/B) or 5s (Aus C)
> * `EnergyDerRecord.protectionMode.overFrequencyProtection`: AS4777-2 default is 52Hz (Aus A/B) or 55Hz (Aus C)
> * `EnergyDerRecord.protectionMode.overFrequencyProtectionDelay`: AS4777-2 has this marked as `-` and it is missing in the DER Guidelines but exists in the DER Register Technical specification
> * `EnergyDerRecord.protectionMode.underVoltageProtection`: AS4777-2 default is 180V
> * `EnergyDerRecord.protectionMode.underVoltageProtectionDelay`: AS4777-2 default is 10s
> * `EnergyDerRecord.protectionMode.overVoltageProtection`: AS4777-2 default is 265V
> * `EnergyDerRecord.protectionMode.overVoltageProtectionDelay`: AS4777-2 default is 1s
> * `EnergyDerRecord.protectionMode.sustainedOverVoltage`: AS4777-2 default is 275V

The default values have been left out in the CDS on purpose after consultations with AEMO.  This is mainly because the default values cannot be assumed as there are scenarios where the distributor/installer may not supply the value to AEMO and it is unknown.  The underlying specs may also change the default values in the future.

> The following fields seem either non-existent or incorrectly defined:
> 
> * `EnergyDerRecord.protectionMode.sustainedOverVoltageDelay`: There is no trip delay specified for `V>>` in the DER Guidelines and AS4777-2 has these marked as `-`. The DER Register Technical specification _has_ this field but it is described as "Sustained Over voltage protection delay in volts (V)" which seems quite wrong (it's a time measurement not a voltage). It's unclear if this field even exists?

The description of the above field in [DER spec](https://aemo.com.au/-/media/files/electricity/nem/it-systems-and-change/2020/der-register-release-schedule-and-technical-specification.pdf),  is defined as - "Sustained Over voltage protection delay in volts (V)". 
However, in [Guide to DER APIs](https://aemo.com.au/-/media/files/electricity/nem/it-systems-and-change/2020/guide-to-derr-apis.pdf?la=en) document it is described as - "Sustained Over voltage protection delay in seconds." 

The DSB can update the description of the field based on AEMO's feedback.


> * `EnergyDerRecord.acConnections[].derDevices[].manufacturer` appears to be mandatory and in fact an enumeration ("pick list") in the DER Guidelines with a comment of _"Definitions align to the approved modules list."_.
> * `EnergyDerRecord.acConnections[].derDevices[]. modelNumber ` appears to be mandatory and in fact an enumeration ("pick list") in the DER Guidelines with a comment of _"Definitions align to the approved modules list."_

The fields were not made ENUM based on consultation with AEMO as the values change frequently (more devices models added) and the data provided by the installer may not be consistent.


> Clarification is requested for `oldest-date` and `newest-date` filters:
> 
> * `oldest-date` filter specifies the default value is _"If absent defaults to newest-date minus 24 months days"_. Is this a simple 24 months subtraction (ie. "today" is 2022-05-15 so `oldest-date` would be `2020-05-15`) that ignores other conditions (length of months, leap years etc) or something more complex?

The above understanding is correct, it is a simple 24 months subtraction.

> * What timezone are `oldest-date` and `newest-date` to be considered at? UTC seems feasible but clarification is important relative to `intervalReads` (see below)
> * `EnergyUsageRead.intervalRead.intervalReads` specifies `specified by readIntervalLength beginning at midnight of readStartDate`. What is "midnight"? I _think_ this is AEST however making it explicit in the description would be helpful to implementers

The `oldest-date` and `newest-date` are filters on a fields in a given payload and align to the fields in that payload. For reads, it should be AEST.  The description of `readStartDate` and `readEndDate` can be updated to specify they are in AEST.

**Comment by perlboy** [comment]
> Based on @OpalRussAEMO's [comment](https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/511#issuecomment-1144297476), the DSB proposes to make the above attributes PositiveInteger. We can update the description to state the acceptable values are 1, 2 and 3.

👍 

> > * `EnergyDerRecord.protectionMode.exportLimitkva`: This field also appears to be incorrectly camelCased as KVA and kVA are interchangeable and KVA is used elsewhere, would be best to be consistent
> The above attribute is represented exactly as it is from AEMOs DER specification similar to other attributes.

This may be the case but the Standards state:
> Field names **MUST** be defined using camel case with the following clarifications

The justification of aligning to a source document is neither a Standard, a camel case clarification nor makes much sense since the DER specification also includes the following statement:
>  Any data submitted by the NSPs that is related to the CEC is accepted to be non case sensitive.

By the above interpretation the attribute name for `BankingTransactionDetail.x2p101Payload.endToEndId` should actually be `BankingTransactionDetail.x2p101Payload.EndToEndId` because that's what it is in the source documents that define an XML Standard.

Suggest the DSB follow its own Standard regardless of source document or there won't actually **be** a Standard when you get to Telco.

> > The following fields have implied defaults if not specified, may be integers and DER Guidelines state `Default values AS4777-2: 2015 section 7.4` which is a paid document so at a minimum having a description that provides them would be useful:
> > * `EnergyDerRecord.protectionMode.underFrequencyProtection`: AS4777-2 default is 47Hz (Aus A/B) or 45Hz (Aus C)
> <snip>
> The default values have been left out in the CDS on purpose after consultations with AEMO. This is mainly because the default values cannot be assumed as there are scenarios where the distributor/installer may not supply the value to AEMO and it is unknown. The underlying specs may also change the default values in the future.

This makes it even **more** important to treat values AEMO does not know as something other than `null` because AS4777-2 specifies defaults and now the community is going to be split between who paid for the AS and who only read the Standards. 

Suggest instead of a `null` or missing AEMO provides a `-1` or `0` value (which isn't valid for any of the `number` fields so will not conflict) and it is designated as "Unknown" or similar.

> The description of the above field in [DER spec](https://aemo.com.au/-/media/files/electricity/nem/it-systems-and-change/2020/der-register-release-schedule-and-technical-specification.pdf), is defined as - "Sustained Over voltage protection delay in volts (V)". However, in [Guide to DER APIs](https://aemo.com.au/-/media/files/electricity/nem/it-systems-and-change/2020/guide-to-derr-apis.pdf?la=en) document it is described as - "Sustained Over voltage protection delay in seconds."
> The DSB can update the description of the field based on AEMO's feedback.

👍 

> The `oldest-date` and `newest-date` are filters on a fields in a given payload and align to the fields in that payload. For reads, it should be AEST. The description of `readStartDate` and `readEndDate` can be updated to specify they are in AEST.

That would be preferred because it is impossible to state that all interval reads will _always_ be in AEST because payloads may not always be served by AEMO (eg. other non-included states may be different or voluntary data holders providing metering data from consumer solar installations).

Also, can I refloat the question around making `unitOfMeasure` an enumeration? There would be a lot of benefit to having this and @OpalRussAEMO already indicated the intent would be to uppercase these values anyway.

**Comment by CDR-API-Stream** [comment]
> To improve consistency, **bold** all Requirement Level keywords usages (e.g. **MUST**, **SHOULD**, ...) throughout the standards.

These changes have been staged for review here: https://github.com/ConsumerDataStandardsAustralia/standards-staging/compare/maintenance/511...maintenance/511-requirement-levels.

Given the large number of changes they have been staged under a separate feature branch off maintenance/511.

**Comment by perlboy** [comment]
> These changes have been staged for review here: [ConsumerDataStandardsAustralia/standards-staging@maintenance/511...maintenance/511-requirement-levels](https://github.com/ConsumerDataStandardsAustralia/standards-staging/compare/maintenance/511...maintenance/511-requirement-levels).
> Given the large number of changes they have been staged under a separate feature branch off maintenance/511.

Can you click the Create PR Request button please so that an inline review can happen. Otherwise this thread is going to get crazy long.

**Comment by CDR-API-Stream** [comment]
Hi @perlboy, done: https://github.com/ConsumerDataStandardsAustralia/standards-staging/pull/195

Given the changes are staged in a separate branch we can discuss the treatment of them in the next Maintenance Iteration call. This review was conducted under the normative standards review and could be considered under that work: https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/blob/master/reviews/2021-05/analysis/analysis-rfc2119-rfc8174-20210519.md

**Comment by perlboy** [comment]
https://github.com/ConsumerDataStandardsAustralia/standards-staging/pull/195#pullrequestreview-1003723299

Repasting what I wrote as a comment in the PR Review:

> There seems to be non 2119 terms used which is not appropriate. Additionally there seems to be excessive 2119 language in use. While this may be a first draft what is proposed makes significant changes to the underlying Standards as they are understood at present. I've walked through as much as I can to provide assistance but this needs a lot more work.
> 
> I think it's appropriate to quote 2119 here as the approach taken by the DSB seems to be one of mass search and replace:
> 
> > Imperatives of the type defined in this memo must be used with care and sparingly. In particular, they MUST only be used where it is actually required for interoperation or to limit behavior which has potential for causing harm (e.g., limiting retransmisssions) For example, they must not be used to try to impose a particular method on implementors where the method is not required for interoperability.
> 
> It's a pretty big line ball as to whether many of the 2119 statements proposed actually meet this requirement or are simply commentary dressed up as Standards.

**Comment by CDR-API-Stream** [comment]
Below is a list of all the changes the DSB will recommend be made to the DER structure discussed within this CR:

- `EnergyDerRecord.availablePhasesCount` - Make the attributes PositiveInteger and update the description to state the acceptable values are 1, 2 and 3.
- `EnergyDerRecord.installedPhasesCount` - Make the attributes PositiveInteger and update the description to state the acceptable values are 1, 2 and 3.
- `EnergyDerRecord.acConnections.derDevices.count` - Make the attribute PositiveInteger
- `EnergyDerRecord.islandableInstallation` - Make the attribute Boolean 
- `EnergyDerRecord.protectionMode.exportLimitkva` - Change it to camelCase - `exportLimitKva`
- `EnergyDerRecord.protectionMode.sustainedOverVoltageDelay` - Based on feedback from AEMO, update description to "Sustained Over voltage protection delay in seconds."
-  `EnergyUsageRead.readStartDate` - Update description to specify it is AEST and assumed to start from 12:00 am. Remove remove reference to time from description
-  `EnergyUsageRead.readEndDate` - Update description to specify it is in AEST and remove reference to time
- `EnergyUsageRead.meterID` - Change to `meterId` (lowercase d)
- `EnergyUsageRead.unitOfMeasure` -  With regards to the request to making this an ENUM and updating description to specify values will be in uppercase, the DSB proposes no change as the DSB is not accountable for the taxonomy and will defer to AEMO specifications.  The DSB will correct the type to [ExternalRef](https://consumerdatastandardsaustralia.github.io/standards/index.html#common-field-types).

**Comment by CDR-API-Stream** [comment]
> Below is a list of all the changes the DSB will recommend be made to the DER structure discussed within this CR:
> 
> * `EnergyDerRecord.availablePhasesCount` - Make the attributes PositiveInteger and update the description to state the acceptable values are 1, 2 and 3.
> * `EnergyDerRecord.installedPhasesCount` - Make the attributes PositiveInteger and update the description to state the acceptable values are 1, 2 and 3.
> * `EnergyDerRecord.acConnections.derDevices.count` - Make the attribute PositiveInteger
> * `EnergyDerRecord.islandableInstallation` - Make the attribute Boolean
> * `EnergyDerRecord.protectionMode.exportLimitkva` - Change it to camelCase - `exportLimitKva`
> * `EnergyDerRecord.protectionMode.sustainedOverVoltageDelay` - Based on feedback from AEMO, update description to "Sustained Over voltage protection delay in seconds."
> * `EnergyUsageRead.readStartDate` - Update description to specify it is AEST and assumed to start from 12:00 am. Remove remove reference to time from description
> * `EnergyUsageRead.readEndDate` - Update description to specify it is in AEST and remove reference to time
> * `EnergyUsageRead.meterID` - Change to `meterId` (lowercase d)
> * `EnergyUsageRead.unitOfMeasure` -  With regards to the request to making this an ENUM and updating description to specify values will be in uppercase, the DSB proposes no change as the DSB is not accountable for the taxonomy and will defer to AEMO specifications.  The DSB will correct the type to [ExternalRef](https://consumerdatastandardsaustralia.github.io/standards/index.html#common-field-types).

These changes have been staged for review: https://github.com/ConsumerDataStandardsAustralia/standards-staging/commit/0cba7f7176bf2943d0457a1bef88496fc78c555f

**Comment by CDR-API-Stream** [comment]
Issue https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/493 has been staged for review: https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues/493#issuecomment-1186741478

**Comment by CDR-API-Stream** [comment]
> > To improve consistency, bold all Requirement Level keywords usages (e.g. MUST, SHOULD, ...) throughout the standards.
>
> These changes have been staged for review here: https://github.com/ConsumerDataStandardsAustralia/standards-staging/compare/maintenance/511...maintenance/511-requirement-levels.

Given the large number of changes they have been staged under a separate feature branch off maintenance/511.

Based on feedback, this change won't be included in release 1.18.0. The feedback has been helpful and further changes to deal with consistency for requirement levels will be progressed in Maintenance Iteration 12.
